pkgs/games/papermc/default.nix

```diff
@@ -1,10 +1,10 @@
 { lib, stdenv, fetchurl, bash, jre }:
 let
   mcVersion = "1.17.1";
-  buildNum = "97";
+  buildNum = "399";
   jar = fetchurl {
     url = "https://papermc.io/api/v1/paper/${mcVersion}/${buildNum}/download";
```

Comment:
```
❯ ./results/papermc/bin/minecraft-server
Exception in thread "main" io.papermc.fourohfour.FourOhFour:

############################################################################################
#                                                                                          #
#                                        Hey there!                                        #
#                                                                                          #
#------------------------------------------------------------------------------------------#
#                                                                                          #
#   The endpoint you downloaded this application from is no longer in service. In order    #
#     to continue using this application, please view our API documentation for details    #
#     on how to access downloads: https://papermc.io/api/docs/                             #
#                                                                                          #
############################################################################################
```
```suggestion
    url = "https://papermc.io/api/v2/projects/paper/versions/${mcVersion}/builds/${buildNum}/downloads/paper-${mcVersion}-${buildNum}.jar";
```

Reply:
Huhh, I'm surprised that they serve a broken jar instead of blocking the download directly. Thank you for catching this!!